public/js/index.js

```diff
@@ -112,16 +115,20 @@ function readNodeProperties() {
 async function loadAllAppNodes() {
     const url = urljoin(window.location.href, 'mongo/node');
     const params = {
-        method:'GET',
+        method: 'GET',
         headers: {
             'Content-Type': 'application/json',
         }
     };
     const res = await fetch(url, params);
     res.json().then(appNodes => {
-        appNodes.forEach(appNode => {
-            addNode(appNode.name, appNode.category, appNode.desc, appNode._id)
-        })
+        //check if node id already exists 
+        if (checkNodeId(appNodes._id) === true) {
+        } else {
+            appNodes.forEach(appNode => {
+                addNode(appNode.name, appNode.category, appNode.desc, appNode._id)
+            })
```

Comment:
Hier musst du wirklich aufpassen: AppNodes ist eine ganze Liste von AppNode-Objekten. Du musst aber jedes einzelne AppNode überprüfen bevor du es hinzutust

Reply:
Ja theoretisch schon, aber darum geht es hier nicht. in Zeile 126 steht
```javascript
appNodes._id
```
Das Problem dabei ist das ```appNodes``` ein Array ist. Dieses Array hat kein Attribut ```_id``` was wir wollen ist die ```_id``` von jedem Element in ```appNodes```. Daher muss die Überprüfung der ```_id``` in die ```forEach``` Schleife, und dort eben nicht ```appNodes``` sondern ```appNode```.
Du kannst die ja mal meine Änderungen pullen und ausprobieren, jetzt funktionierts.